BoostTestAdapter/Utility/CommandLine.cs

```diff
@@ -137,13 +147,55 @@ public override string ToString()
         /// <returns>The command line instance parsed from cmdLine</returns>
         public static CommandLine FromString(string cmdLine)
         {
-            cmdLine = (cmdLine == null) ? string.Empty : cmdLine;
+            cmdLine = cmdLine ?? string.Empty;
 
             var splitCommandLine = CommandLineArgExtensions.SplitCommandLine(cmdLine);
             return new CommandLine(
                 splitCommandLine.FirstOrDefault(),
                 splitCommandLine.Skip(1)
             );
         }
+
+        /// <summary>
+        /// Concatenates a collection of strings as a valid command-line argument set
+        /// </summary>
+        /// <param name="arguments">The arguments to serialize</param>
+        /// <returns>string representing the concatenation set of all arguments</returns>
+        private static string JoinArguments(IEnumerable<string> arguments)
+        {
+            if (arguments == null)
+            {
+                return string.Empty;
+            }
+
+            var quotedArgs = arguments.Select(arg => arg.Contains(' ') ? string.Format(CultureInfo.InvariantCulture, "\"{0}\"", arg) : arg);
+            return string.Join(" ", quotedArgs);
+        }
+
+        /// <summary>
+        /// Normalizes the file path and adds quotes should the path not be quoted already
+        /// </summary>
+        /// <param name="filePath">The path to notmalized</param>
+        /// <returns>A normalized file path</returns>
+        private static string NormalizePath(string filePath)
```

Comment:
The usual meaning of path normalization is somewhat wider than what this function does [[1](https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/mitigation-path-normalization#what-is-path-normalization)]. I suggest renaming this method to avoid confusion.